+bids/layout.m

```diff
@@ -365,36 +378,56 @@
   pth = fullfile(subject.path, modality);
   fullpath_filename = fullfile(pth, file);
 
-  subject.(modality)(end).metafile = bids.internal.get_meta_list(fullpath_filename);
-  subject.(modality)(end).dependencies.explicit = {};
-  subject.(modality)(end).dependencies.data = {};
-  subject.(modality)(end).dependencies.group = {};
+  if ~isfield(subject.(modality)(end), 'dependencies') || ...
+          isempty(subject.(modality)(end).dependencies)
+    subject.(modality)(end).dependencies.explicit = {};
+    subject.(modality)(end).dependencies.data = {};
+    subject.(modality)(end).dependencies.group = {};
+  end
```

Comment:
The creation of dependencies field should be defined where the structure `subject.(modality)(end)` is created (i.e. in `append_to_layout` function). It will assure that subjects structure remains consistent.